src/Compilers/Core/Portable/MetadataReference/AssemblyMetadata.cs

```diff
@@ -64,10 +64,7 @@ public bool IsDisposed
         /// <summary>
         /// Cached assembly symbols.
         /// </summary>
-        /// <remarks>
-        /// Guarded by <see cref="CommonReferenceManager.SymbolCacheAndReferenceManagerStateGuard"/>.
-        /// </remarks>
-        internal readonly WeakList<IAssemblySymbol> CachedSymbols = new WeakList<IAssemblySymbol>();
+        internal readonly CachedAssemblySymbolList CachedSymbols = new CachedAssemblySymbolList();
```

Comment:
>= new CachedAssemblySymbolList() [](start = 65, length = 32)

Consider moving this to an `else` in the private constructor and explicitly initializing in the other constructors to avoid generating unnecessary garbage.